code/modules/wiremod/component_printer.dm

```diff
@@ -1,3 +1,118 @@
+/// Component printer, creates components for integrated circuits.
 /obj/machinery/component_printer
-	name = "Component Printer"
-	desc = "Huge pile of nothing"
+	name = "component printer"
+	desc = "Produces components for the creation of integrated circuits."
+	// MOTHBLOCKS TODO: We need a real icon
```

Comment:
!

Reply:
We need icons for some other stuff as well, will either just whip up some crap, loot from unused sprites, or ask someone.